provisioner/src/lib.rs

```diff
@@ -236,6 +261,269 @@ impl MyProvisioner {
+            Ok(_) => {}
+            Err(e) => {
+                match e.into_service_error() {
+                    CreatePolicyError::EntityAlreadyExistsException(_) => {} // for idempotency
+                    e => {
+                        return Err(Error::CreateIAMPolicy(e));
+                    }
+                }
+            }
+        };
+
+        Ok(())
+    }
+
+    async fn get_policy_arn(&self, prefix: &str) -> Result<String, Error> {
+        let identity = self.sts_client.get_caller_identity().send().await.map_err(|e| Error::GetCallerIdentity(e))?;
+        let account = identity.account().ok_or_else(|| Error::GetAccount("empty account".to_string()))?;
+
+        let policy_name = self.get_dynamodb_policy_name(prefix).await;
+        let policy_arn = format!("arn:aws:iam::{account}:policy/{policy_name}");
+
+        Ok(policy_arn)
+    }
+
+    async fn delete_dynamodb_policy(&self, prefix: &str) -> Result<(), Error> {
+        let policy_arn = self.get_policy_arn(&prefix).await?;
+
+        self.iam_client
+            .delete_policy()
+            .policy_arn(policy_arn)
+            .send()
+            .await
+            .map_err(|e| Error::DeleteIAMPolicy(e))?;
+
+        Ok(())
+    }
+
+    async fn get_iam_identity_user_name(&self, prefix: &str) -> String {
+        // max characters is 64
+        format!("{}-dynamo-user", prefix)
+    }
+
+    async fn create_iam_identity(&self, prefix: &str) -> Result<(), Error> {
+        match self
+            .iam_client
+            .create_user()
+            .user_name(self.get_iam_identity_user_name(prefix).await)
+            .send()
+            .await {
+                Ok(_) => {},
+                Err(e) => {
+                    match e.into_service_error() {
+                        CreateUserError::EntityAlreadyExistsException(_) => {}
+                        e => {
+                            return Err(Error::CreateIAMUser(e));
+                        }
+                    }
+                }
+            };
+        Ok(())
+    }
+
+    async fn get_saved_access_key(&self, prefix: &str) -> Option<(String, String)> {
+        if let Ok(file) = std::fs::File::open(self.get_access_key_file_name(prefix)) {
+            let mut lines = std::io::BufReader::new(file).lines();
+
+            if let Some(Ok(access_key_id)) = lines.next() {
+                if let Some(Ok(secret_access_key)) = lines.next() {
+                    return Some((access_key_id, secret_access_key))
+                }
+            }
+        }
+
+        None
+    }
+
+    fn get_access_key_file_name(&self, prefix: &str) -> String {
+        format!("{}.txt", prefix)
+    }
+
+    async fn delete_saved_access_key(&self, prefix: &str) -> Result<(), std::io::Error> {
+        std::fs::remove_file(self.get_access_key_file_name(prefix))?;
+        Ok(())
+    }
+
+    async fn save_access_key(&self, prefix: &str, access_key_id: &str, secret_access_key: &str) -> Result<(), std::io::Error> {
```

Comment:
Question for Shuttle devs: where and how would you like us to store these access keys?

Currently we're using .txt files. Presumably that's not the desired end solution though.